lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -2964,11 +2965,49 @@ void ByteCodeGenerator::EmitOneFunction(ParseNode *pnode)
 
         DefineLabels(funcInfo);
 
-        if (pnode->sxFnc.HasNonSimpleParameterList() || !funcInfo->IsBodyAndParamScopeMerged())
+        // We need to emit the storage for special symbols before we emit the default arguments in case the default
+        // argument expressions reference those special names.
+        if (pnode->sxFnc.HasNonSimpleParameterList())
         {
-            Assert(pnode->sxFnc.HasNonSimpleParameterList() || CONFIG_FLAG(ForceSplitScope));
+            // If the param and body scope are merged, the special symbol vars are located in the body scope so we
+            // need to walk over the var list.
+            if (funcInfo->IsBodyAndParamScopeMerged())
+            {
+                for (ParseNodePtr pnodeVar = pnode->sxFnc.pnodeVars; pnodeVar; pnodeVar = pnodeVar->sxVar.pnodeNext)
+                {
+#if DBG
+                    bool reachedEndOfSpecialSymbols = false;
+#endif
+                    Symbol* sym = pnodeVar->sxVar.sym;
 
-            if (paramScope)
+                    if (sym != nullptr && sym->IsSpecialSymbol())
```

Comment:
>sym != nullptr [](start = 24, length = 14)

With this change can it ever be null?

Reply:
Yes, I think so. The sym could still be a non-special symbol of any flavor.